File: JAVA/1408. String Matching in an Array.java
class Solution {
    public List<String> stringMatching(String[] w) {
        Set<String> a=new HashSet<>();
        for(int i=0;i<w.length;i++){
            for(int j=i+1;j<w.length;j++){
                if(w[i].length()>w[j].length()){
                    if(w[i].contains(w[j]))
                    a.add(w[j]);
                }
                else{
                    if(w[j].contains(w[i]))
                    a.add(w[i]);
                }
        }
        }
        return new ArrayList<>(a);
}
}

File: C++/1408. String Matching in an Array.cpp
class Solution {
public:
    vector<string> stringMatching(vector<string>& w) {
        set<string>s;
        for(int i=0;i<w.size();i++){
            for(int j=i+1;j<w.size();j++){
                if(w[i].size()>w[j].size()){
                    if(w[i].find(w[j])!=-1)
                    s.insert(w[j]);
                }
                else{
                    if(w[j].find(w[i])!=-1)
                    s.insert(w[i]);
                }
            }
        }
        vector<string>a(s.begin(),s.end());
        return a;
    }
};